pkg/cmd/agentversionjob.go

```diff
@@ -1,118 +1,49 @@
 package cmd
 
 import (
-	"encoding/json"
+	"fmt"
 
+	"github.com/Axway/agent-sdk/pkg/agent"
+	"github.com/Axway/agent-sdk/pkg/agent/resource"
+	management "github.com/Axway/agent-sdk/pkg/apic/apiserver/models/management/v1alpha1"
 	"github.com/Axway/agent-sdk/pkg/config"
 	"github.com/Axway/agent-sdk/pkg/util"
 
 	"regexp"
-	"strconv"
 	"strings"
 
-	"net/http"
-
-	coreapi "github.com/Axway/agent-sdk/pkg/api"
 	"github.com/Axway/agent-sdk/pkg/jobs"
 	"github.com/Axway/agent-sdk/pkg/util/errors"
 	log "github.com/Axway/agent-sdk/pkg/util/log"
 )
 
 const (
 	avcCronSchedule = "@daily"
-	jfrogURL        = "https://axway.jfrog.io/ui/api/v1/ui/treebrowser"
 )
 
-var agentNameToRepoPath = map[string]string{
-	"AWSDiscoveryAgent":                      "aws-apigw-discovery-agent",
-	"AWSTraceabilityAgent":                   "aws-apigw-traceability-agent",
-	"AzureDiscoveryAgent":                    "azure-discovery-agent",
-	"AzureTraceabilityAgent":                 "azure-traceability-agent",
-	"EnterpriseEdgeGatewayDiscoveryAgent":    "v7-discovery-agent",
-	"EnterpriseEdgeGatewayTraceabilityAgent": "v7-traceability-agent",
-}
-
-type version struct {
-	major, minor, patch int
-	val                 string
-}
-
-type jfrogRequest struct {
-	Type     string `json:"type"`
-	RepoType string `json:"repoType"`
-	RepoKey  string `json:"repoKey"`
-	Path     string `json:"path"`
-	Text     string `json:"text"`
-}
-
-type jfrogData struct {
-	Items []jfrogItem `json:"data"`
-}
-
-type jfrogItem struct {
-	RepoKey      string `json:"repoKey,omitempty"`
-	Path         string `json:"path,omitempty"`
-	Version      string `json:"text"`
-	RepoType     string `json:"repoType,omitempty"`
-	HasChild     bool   `json:"hasChild,omitempty"`
-	Local        bool   `json:"local,omitempty"`
-	Type         string `json:"type,omitempty"`
-	Compacted    bool   `json:"compacted,omitempty"`
-	Cached       bool   `json:"cached,omitempty"`
-	Trash        bool   `json:"trash,omitempty"`
-	Distribution bool   `json:"distribution,omitempty"`
-}
-
 // AgentVersionCheckJob - polls for agent versions
 type AgentVersionCheckJob struct {
```

Comment:
lets add a field logger here like we do for other packages and use it through out this job